### ISSUE TITLE
[Bug] Rerender Error - Uncaught Error: Window closed

### ISSUE THREAD

**Issue by Manuel83** [paypal/react-paypal-js#45]
### 🐞 Describe the Bug

When the PayPalButtons Component is used in a Form which is again after state change  the javascript console shows an error. 

![image](https://user-images.githubusercontent.com/5859486/96916995-94e48e80-14a8-11eb-8c8b-3dc71bdb5b9f.png)

### 🔬 Minimal Reproduction
![image](https://user-images.githubusercontent.com/5859486/96918031-f35e3c80-14a9-11eb-852a-01e736663b0a.png)

`  

    import React from "react";
    import ToggleButton from '@material-ui/lab/ToggleButton';
    import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';
    import {PayPalButtons, PayPalScriptProvider} from "@paypal/react-paypal-js";

    const ShopView = () => {

    const [amount, setAmount] = React.useState('0');

    const handleAlignment = (event, newAlignment) => {

        if (newAlignment) {
            setAmount(newAlignment);
        }

    };

    return (<div>
        <PayPalScriptProvider options={{currency: "EUR", "client-id": ".... YOUR CLIENT ID GOES HEERE... "}}>

            <ToggleButtonGroup
                size="small"
                value={amount}
                exclusive
                onChange={handleAlignment}
                aria-label="text alignment"
            >
                <ToggleButton value="4" aria-label="left aligned">
                    Amount 4 Euro
                </ToggleButton>
                <ToggleButton value="2" aria-label="centered">
                    Amount 2 Euro
                </ToggleButton>
            </ToggleButtonGroup>

            <PayPalButtons/>

        </PayPalScriptProvider>

    </div>)
    }

     export default ShopView

`

### 😕 Actual Behavior

When toggeling the the amounts an error is thrown. The error only ourres when I'm toggling fast. 
Each time the paypal script gets reloaded. If I update the component state which will force a rerender before the Paypal script is loaded the error is thrown.

### 🤔 Expected Behavior

No Error is thrown in the JS Console

### 🌍 Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-paypal-js  | 1.0.1
| Browser          | Chrome
| Operating System | MacOS

**Comment by sajadghawami** [comment]
Having the same problem

**Comment by gregjopa** [comment]
@Manuel83 thanks for creating the issue. This is indeed a bug with this react component. I am able to reproduce this same error with the [Currency demo](https://paypal.github.io/react-paypal-js/?path=/story/example-usepaypalscriptreducer--currency-usd) when quickly switching options.

@sajadghawami I believe this bug is related to this zoid issue: https://github.com/krakenjs/zoid/issues/218. The key learning from this issue is that `render()` is an async function and the error needs to be caught. I'm working on a fix to catch this error and log it like so:

```js
buttons.current.render(buttonsContainerRef.current).catch((err) => {
    console.error("Failed to render buttons component", err);
});
```

**Comment by sajadghawami** [comment]
@braebot thanks for the quick reply.

i encountered that `zoid issue` too, but was not sure where it came from.

**Comment by gregjopa** [comment]
I just published version 1.0.3 which contains the fix from #47 to catch this render error and log it using `console.error()`.
